packages/utils/src/lib/merge-configs.ts

```diff
@@ -0,0 +1,229 @@
+import {
+  CategoryConfig,
+  CoreConfig,
+  Group,
+  GroupRef,
+  PersistConfig,
+  PluginConfig,
+  UploadConfig,
+} from '@code-pushup/models';
+
+export function mergeConfigs(
+  config: CoreConfig,
+  ...configs: Partial<CoreConfig>[]
+) {
```

Comment:
We could type the return type.. e.g. 
```suggestion
): Record<'categories', CategoryConfig[]> {
```

Reply:
Agreed, although the accurate type would be:

```suggestion
): Pick<CoreConfig, 'categories'> {
```